server/app/auth/CiviFormProfileMerger.java

```diff
@@ -0,0 +1,69 @@
+package auth;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+import javax.inject.Provider;
+import models.Account;
+import models.Applicant;
+import org.pac4j.core.profile.UserProfile;
+import repository.UserRepository;
+
+/** This class serves to abstract out common user profile merging logic into a shared helper. */
+public class CiviFormProfileMerger {
+
+  private final ProfileFactory profileFactory;
+  private final Provider<UserRepository> applicantRepositoryProvider;
+
+  public CiviFormProfileMerger(
+      ProfileFactory profileFactory, Provider<UserRepository> applicantRepositoryProvider) {
+    this.profileFactory = profileFactory;
+    this.applicantRepositoryProvider = applicantRepositoryProvider;
+  }
+
+  /**
+   * This method handles a three way merge between an existing applicant, an existing guest profile,
+   * and an external profile
```

Comment:
We follow the google java and javadoc styleguides https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

So this summary line should be more like this. a fragment with ending punctuation.

\* Merges multiple profile sources into a {@code UserProfile}.